code/controllers/subsystem/timer.dm

```diff
@@ -128,6 +128,10 @@ SUBSYSTEM_DEF(timer)
 		callBack.InvokeAsync()
 
 		if(ctime_timer.flags & TIMER_LOOP)
+			if (QDELETED(ctime_timer))
+				if (MC_TICK_CHECK) // to prevent overrun from callbacks running
+					break
+				continue
```

Comment:
tick check isn't needed because of where continue lands. also pls add a comment explaining why this is done thanks